### PR TITLE
Add Suspend option to Quit menu

### LINKED ISSUE #720: Add option to suspend system
Hi, would it be possible to add the option to _Suspend_ the system on top of the current _Shutdown_ and _Reboot_?
I'm starting to set up a system using an Nvidia Shield with Steam Link connected to a Windows PC in another room. I would like to be able to put to sleep the PC once finished gaming. I haven't tested it yet when Steam is minimised but it should be able to wake from sleep the same way it does when in Big Picture mode. Also, there is the option to start in Desktop mode so that way I should be able to leave Pegasus open all the time and retake the session once the PC is woken up.
Thanks.

### PR DESCRIPTION
**Pull request by j8r**
Resolves #720, at least for Linux. Works, but didn't add Windows/macOS support, because I can't test, and not familiar with this platforms (also, I didn't find much _good_ docs).

My first time with C++/Qt and this project, likely a lot of things are missing or not done right.

Notes:
- One little issue is the menu is still present when going back from sleep, ideally we should just see the main menu with games.
- I didn't put a confirmation dialog. I find it more annoying than useful, suspend is a fast and quite safe operation, in case of a mistake.

Side question: I find it inconvenient to have translations in a submodule, what's the advantage over putting it in this repository directly? It will be easier to clone and contribute (one less fork and PR).

### CODE REVIEW

**Comment by mmatyas** [comment]
Thanks for the patch! The code looks OK, though you'll still need at least a dummy Suspend function for Windows too (or, even better, https://docs.microsoft.com/en-us/windows/win32/api/powrprof/nf-powrprof-setsuspendstate). I also wonder why the GitHub pull request checks didn't run, hmm...

The advantage of a translations repo is that it avoids spamming the main repo with small changes, like typo fixes, allows updating languages in bulk, allows separating language-specific issues and workflow from the main repo, and allows separate permission controls, eg. in case we add a translation service that modifies the files of a repo. I'm fine with it.

**Comment by j8r** [comment]
> The advantage of a translations repo is that it avoids spamming the main repo with small changes, like typo fixes, allows updating languages in bulk, allows separating language-specific issues and workflow from the main repo, and allows separate permission controls, eg. in case we add a translation service that modifies the files of a repo. I'm fine with it.

Maybe, but currently there are 2 issues, 40 commits, no translation tool, and I guess you're the only owner. Anyway that's your project, do as you wish.

In the translation files, what correspond the `line` in for example `<location filename="../src/frontend/menu/MainMenuPanel.qml" line="96"/>`. Sometimes there is nothing to translate at this line.

Also, do you have an idea how to get rid of the menu after going back from sleep?

**Comment by j8r** [comment]
> you'll still need at least a dummy Suspend function for Windows too

I thought it was the purpose of `PowerCommands_unimpl.cpp`?

**Comment by mmatyas** [comment]
For the translation, you can use Qt Linguist, which shows the source too and can update the files. Take a look at https://pegasus-frontend.org/docs/dev/translate/. Alternatively, if the files are not updated, it will just stay in English for now.

> Also, do you have an idea how to get rid of the menu after going back from sleep?

There should be a function or `close` call, like when you close the menu naturally. It should be possible to call that after the suspend command.

> I thought it was the purpose of PowerCommands_unimpl.cpp?

That file is for completely unimplemented platforms, like Android. As all the remaining platforms support both reboot and shutdown, there's no fallback mechanism. You could copy the functions to the Windows/Mac ones for now, as a temporary workaround.

**Comment by mmatyas** [comment]
Meanwhile I've also fixed the build checks not triggering. If you rebase to master, they should now launch correctly, I hope.

**Comment by j8r** [comment]
The Android CI fails for an unrelated reasons, at the `openssl aes-256-cbc -k "${AKS_ENC_KEY}" -in aks.enc -out aks -d` stage.

File: src/backend/Backend.cpp [hunk 1]
```diff
@@ -94,6 +94,10 @@ void register_api_classes()
 
 void on_app_close(AppCloseType type)
 {
+    if (type == AppCloseType::SUSPEND) {
+        // close the menu
```
Comment (mmatyas) [inline src/backend/Backend.cpp:98]: Here ^

**Review by mmatyas — COMMENTED** [review]
Ok, looks fine! Don't worry about Android, will look into the script later.

Just one thing: you've added support for on-suspend script, but you haven't called `ScriptRunner::run` (eg. at the place I've marked). But thinking about it, that might also have some complication: if one want to have a script that should run _before_ suspend, eg. to pause some service, then we would probably need a support for scripts to run _after_ the suspend ends too.

Since this might mean some more changes, to keep the commit small, for now I'd recommend removing the changes to the ScriptRunner code. Of course, if you'd like implement such feature, I'm happy to see it in a follow up pull request.

I was thinking about the README change too, but it's not really a requirement of the "Add suspend option" feature, so that could be dropped too. The documentation is located here: https://github.com/mmatyas/pegasus-docs/blob/master/docs/dev/build.md, which is where the main build documentation is, if you'd like to edit it.

**Comment by j8r** [comment]
Yes I can do another PR for this little doc change.

**Comment by j8r** [comment]
All should be good now.

**Comment by mmatyas** [comment]
Thanks!

**Pull request merged**